sencha-workspace/SlateTasksStudent/app/controller/Dashboard.js

```diff
@@ -136,6 +148,49 @@ Ext.define('SlateTasksStudent.controller.Dashboard', {
         });
     },
 
+    onStudentSelectorSelect: function(combo, rec) {
+        var me = this,
+            studentCombo = me.getStudentSelectorCombo(),
+            params = me.getSearchParams(),
+            username;
+
+        if (!studentCombo) {
+            return;
+        }
+
+        // skip rerouting if student selection hasn't changed
+        username = studentCombo.getValue();
+        if (username === params.student) {
+            return;
+        }
+
+        // reroute page to reflect selected student
+        params.student = username;
+        location.search = Ext.Object.toQueryString(params);
+    },
+
+    onStudentSelectorBoxReady: function(combo) {
+        var params = this.getSearchParams();
+        combo.getStore().load();        
```

Comment:
@jtwiest `load()` is an asynchronous call, if you're relying on it this is a race condition. You could pass `{ callback: function() { ... } }` to it if you needed something to be done after it was loaded. You don't usually need to manually trigger combo box stores to load though, and we don't want it loading all people in the system with no query. Just set the value and let combobox worry about when it loads

Reply:
@themightychris I tried doing it without the load and it's not working b\c we're using a custom display tpl. In other words we're setting the value as the Username field but rendering the First / Last name fields.

I used the load callback method to populate this value instead.